### PR TITLE
Cook/add new model setup

### PR DESCRIPTION
**Pull request by mandresm**
Sections moved from the `Contributing` chapter to the `Cookbook`.

An update was needed on these sections to suit release 4.2 changes.

See the resulting recipe here:
- https://esm-tools.readthedocs.io/en/cook-add_new_model_setup/cookbook.html#implement-a-new-model
- https://esm-tools.readthedocs.io/en/cook-add_new_model_setup/cookbook.html#implement-a-new-coupled-setup

### CODE REVIEW

File: docs/recipes/add_model_setup.rst [hunk 1]
```diff
@@ -30,96 +30,190 @@ Implement a New Model
...
+   and create the model's `yaml` file::
 
     $ mkdir <PATH>/esm_tools/configs/<model>
-    $ touch <PATH>/esm_tools/configs/<model>/<model-version>.yaml
+    $ touch <PATH>/esm_tools/configs/<model>/<model>.yaml
 
-   These `yaml` files need to exist for `esm_master` to download and compile your model, but they can
-   be empty. However, you can choose to fill them with a basic configuration::
+5. Use your favourite text editor to open and edit your ``<model>.yaml`` in the
+   ``esm_tools/configs/components/model`` folder::
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: Line 43: 
`esm_tools/configs/components/model`
should be
`esm_tools/configs/components/<model>/`

File: docs/recipes/add_model_setup.rst [hunk 2]
```diff
@@ -30,96 +30,190 @@ Implement a New Model
...
 
-    # YOUR_MODEL YAML CONFIGURATION FILE
-    #
+    $ <your_text_editor> <PATH>/esm_tools/configs/components/<model>.yaml
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: Line 45: In the path there is a <model> folder missing:
```
$ <your_text_editor> <PATH>/esm_tools/configs/components/<model>.yaml
$ <your_text_editor> <PATH>/esm_tools/configs/components/<model>/<model>.yaml
```

File: docs/recipes/add_model_setup.rst [hunk 3]
```diff
@@ -30,96 +30,190 @@ Implement a New Model
...
+   Then you can actually try installing your model in the desired folder::
 
     $ mkdir ~/model_codes
     $ cd ~/model_codes
     $ esm_master install-your_model-version
 
-8. If everything works correctly you can check that your changes pass `flake8`, commit your changes, push
-   them to the ``origin`` and submit a pull request through GitHub (see steps 5-7 in
-   :ref:`contributing:Contribution to esm_tools Package`).
+10. If everything works correctly you can check that your changes pass ``flake8``, commit your changes, push
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: Line 201: We could give a small example of how to use `flake8`:
`flake8 path/to/code/`

And maybe link `flake8` to a documentation https://flake8.pycqa.org/en/latest/index.html#

File: docs/recipes/add_model_setup.rst [hunk 4]
```diff
@@ -222,84 +222,122 @@ Implement a New Coupled Setup
...
-3. Then you will need to create a folder for your coupled setup inside ``esm_tools/configs/setups``
-   folder, and create a `yaml` file per version of your setup::
+3. Setups need two types of files: 1) **coupling files** containing information about model versions and
+   coupling changes, and 2) **setup files** containing the general information about the setup and the
+   model changes. In this step we focus on the creation of the **coupling files**.
+
+   a. Create a folder for your setup's couplings in ``esm_tools/configs/couplings``, and inside this
+      folder add a folder per coupling::
+
+       $ mkdir esm_tools/configs/couplings/<your_setup>
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: I think there is no folder for the setup within the `couplings` folder. All the coupling folders go directly into `esm_tools/configs/couplings/`.

Line 242: `$ mkdir esm_tools/configs/couplings/<your_setup>` not needed.
Line 243: `$ cd esm_tools/configs/couplings/<your_setup>` can be
`$ cd esm_tools/configs/couplings/`

File: docs/recipes/add_model_setup.rst [hunk 5]
```diff
@@ -222,84 +222,122 @@ Implement a New Coupled Setup
...
+         - [ ... ]
+         - "coupler-version"
+         coupling_changes:
+         - sed -i '/MODEL1_PARAMETER/s/OFF/ON/g' model1-1.0/file_to_change
+         - sed -i '/MODEL2_PARAMETER/s/OFF/ON/g' model2-1.0/file_to_change
+         - [ ... ]
+
+      The ``components`` section should list the models and couplers used for the given coupling
+      including their required version. The ``coupling_changes`` subsection should include a list of
+      commands to make the necessary changes in the component's make files, for a correct compilation
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: Line 271: Is the second make actually the `make` command?

File: docs/recipes/add_model_setup.rst [hunk 6]
```diff
@@ -310,9 +348,83 @@ Implement a New Coupled Setup
     $ cd ~/model_codes
     $ esm_master install-your_setup-version
 
-8. If everything works correctly you can check that your changes pass `flake8`, commit your changes, push
-   them to the ``origin`` and submit a pull request through GitHub (see steps 5-7 in
-   :ref:`contributing:Contribution to esm_tools Package`).
+10. If everything works correctly you can check that your changes pass ``flake8``, commit your changes, push
+    them to the ``origin`` and submit a pull request through GitHub (see steps 5-7 in
```
Comment (nwieters) [inline docs/recipes/add_model_setup.rst]: Line 351/352: I would add here something like "push them to the `origin` remote repository and ..."

File: docs/recipes/add_model_setup.rst (hunk 4 above)
Comment (mandresm) [reply docs/recipes/add_model_setup.rst]: You are totally right, thanks for the heads-up.

**Pull request merged**